pkg/executor/remote.go

```diff
@@ -366,62 +423,86 @@ func (ex *Remote) sftpDownload(ctx context.Context, req sftpReq) error {
 	}
 	defer sftpClient.Close()
 
-	remoteFh, err := sftpClient.Open(req.remoteFile)
+	matches, err := sftpClient.Glob(req.remoteFile)
```

Comment:
the whole change here is misplaced. sftDownload is the wrong abstraction level to perform all of this higher-level matching logic. As you can see, other methods do it on a higher level, i.e. Sync. Delete and so on.

Reply:
You are right, my bad. I've moved the glob-exclude logic to the level of the Download function.